content/chainguard/chainguard-enforce/iam-groups/identity-examples/enforce-buildkite-identity.md

```diff
@@ -0,0 +1,420 @@
+
+```sh
+cat > buildkite.tf <<EOF
+resource "chainguard_identity" "buildkite" {
+  parent_id   = chainguard_group.user-group.id
+  name    	= "buildkite"
+  description = <<EOF
+	This is an identity that authorizes Buildkite workflows
+	for this repository to assume to interact with chainctl.
+  EOF
+
+  claim_match {
+	issuer      	= "https://agent.buildkite.com"
+	subject_pattern = "organization:<organization-name>:pipeline:<pipeline-name>:ref:refs/heads/main:commit:[0-9a-f]+:step:"
+  }
+}
+
+output "buildkite-identity" {
+  value = chainguard_identity.buildkite.id
+}
+
+data "chainguard_roles" "viewer" {
+  name = "viewer"
+}
+
+resource "chainguard_rolebinding" "view-stuff" {
+  identity = chainguard_identity.buildkite.id
+  group	= chainguard_group.user-group.id
+  role 	= data.chainguard_roles.viewer.items[0].id
+}
+EOF
+```
+
+Following that, your Terraform configuration will be ready. Now you can run a few `terraform` commands to create the resources defined in your `.tf` files.
+
+
+## Creating Your Resources
+
+First, run `terraform init` to initialize Terraform's working directory.
+
+```sh
+terraform init
+```
+
+Then run `terraform plan`. This will produce a speculative execution plan that outlines what steps Terraform will take to create the resources defined in the files you set up in the last section.
+
+```sh
+terraform plan
+```
+
+If the plan worked successfully and you're satisfied that it will produce the resources you expect, you can apply it.
+
+```sh
+terraform apply
+```
+
+Before going through with applying the Terraform configuration, this command will prompt you to confirm that you want it to do so. Enter `yes` to apply the configuration.
+
+```
+. . .
+
+Plan: 3 to add, 0 to change, 0 to destroy.
+
+Changes to Outputs:
+  + buildkite-identity = (known after apply)
+
+Do you want to perform these actions?
+  Terraform will perform the actions described above.
+  Only 'yes' will be accepted to approve.
+
+  Enter a value:
+```
+
+After pressing `ENTER`, the command will complete and will output an `buildkite-identity` value.
+
+```
+. . .
+
+Apply complete! Resources: 3 added, 0 changed, 0 destroyed.
+
+Outputs:
+
+buildkite-identity = "<your buildkite identity>"
+```
+
+This is the identity's UIDP (unique identity path), which you configured the `buildkite.tf` file to emit in the previous section. Note this value down, as you'll need it when you test this identity using a Buildkite workflow. If you need to retrieve this UIDP later on, though, you can always run the following `chainctl` command to obtain a list of the UIDPs of all your existing identities.
```

Comment:
```suggestion
This is the identity's [UIDP (unique identity path)](/chainguard/chainguard-enforce/reference/events/#uidp-identifiers), which you configured the `buildkite.tf` file to emit in the previous section. Note this value down, as you'll need it when you test this identity using a Buildkite workflow. If you need to retrieve this UIDP later on, though, you can always run the following `chainctl` command to obtain a list of the UIDPs of all your existing identities.
```

Reply:
Just adding this here for now, though I think I should split that UIDP thing into a separate reference page at some point

Reply:
thanks for this! I added it to the github actions example too